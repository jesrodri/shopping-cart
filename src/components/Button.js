import styled from "styled-components";

const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.buttonBackground.default};
  color: ${({ theme }) => theme.colors.buttonText.default};
  border-radius: ${({ theme }) => theme.radii[2]};
  min-width: ${({ theme }) => theme.sizes[6]};
  font-weight: ${({ theme }) => theme.fontWeights[3]};
  font-size: ${({ theme }) => theme.fontSizes.body2};
  outline: 0;
  border: 0;
  margin: 0;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  padding: 6px 16px;
  margin-top: 24px;
  margin-left: 8px;

  &:hover {
    background-color: ${({ theme }) => theme.colors.buttonBackground.hover};
  }

  &:disabled {
    background-color: ${({ theme }) => theme.colors.buttonBackground.disabled};
    color: ${({ theme }) => theme.colors.buttonText.disabled};
  }
`

export default Button;
