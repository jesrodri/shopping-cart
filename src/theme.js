const colors = {
  black:       '#000000',
  white:       '#ffffff',
  lightGray:   '#b5b0a7',
  gray:        '#999999',
  darkGray:    '#37352f',
  darkRed:     '#cc0000',
  yellow:      '#e0ca00',
  lightYellow: '#f7e581',
  salmon:      '#fa8072',
  darkSalmon:  '#f16050',
  green:       '#66f57e',
  darkGreen:   '#49a659',
}

const space = [0, 4, 8, 16, 32, 64, 128, 256, 512]

const fonts = {
  body:
    "-apple-system, BlinkMacSystemFont, 'Segoe UI', Helvetica, 'Apple Color Emoji', Arial, sans-serif, 'Segoe UI Emoji', 'Segoe UI Symbol', 'Nunito', sans-serif;",
}

const fontSizes = [0, 5, 7, 9, 12, 16, 21, 28, 37, 50, 67, 89, 119, 159].map(
  (n) => `${n / 10}rem`
)

const fontWeights = [0, 300, 400, 700]

const radii = [0, 2]

const zIndices = []

const sizes = []

const theme = {
  colors,
  space,
  fonts,
  fontSizes,
  fontWeights,
  radii,
  zIndices,
  sizes,
}

export default theme
