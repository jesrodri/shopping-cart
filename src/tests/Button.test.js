import renderer from 'react-test-renderer';
import Button from '../components/Button';
import { ThemeProvider } from 'styled-components';
import theme from '../theme';

describe('<Button />', () => {
  it('renders Button correctly', () => {
    const button = renderer.create(
      <ThemeProvider theme={theme}>
        <Button>Button</Button>
      </ThemeProvider>
    ).toJSON();
    
    expect(button).toMatchSnapshot();
  });
});
