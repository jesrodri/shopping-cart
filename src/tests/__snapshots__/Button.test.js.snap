// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders Button correctly 1`] = `
<button
  className="Button-sc-2h6cpf-0 gFjMGv"
>
  Button
</button>
`;
